src/Components/Web/src/JSComponents/JSComponentInterop.cs

```diff
@@ -93,11 +96,36 @@ protected internal void SetRootComponentParameters(int componentId, int paramete
                 object? parameterValue;
                 if (TryGetComponentParameterType(componentType, parameterName, out var parameterType))
                 {
-                    // It's a statically-declared parameter, so we can parse it into a known .NET type
-                    parameterValue = JsonSerializer.Deserialize(
-                        parameterJsonValue,
-                        parameterType,
-                        jsonOptions);
+                    // It's a statically-declared parameter, so we can parse it into a known .NET type.
+
+                    if (parameterType == typeof(EventCallback))
```

Comment:
As a way of centralizing these decisions about whether something is func-typed and what kind of func typed, what do you think about refactoring the `JSComponentInterop.ParameterTypeCache.ParameterTypes` information to contain this?

For example, changing:

```cs
public readonly Dictionary<string, Type> ParameterTypes;
```

... to something that contains more info like:

```cs
public readonly Dictionary<string, ParameterTypeInfo> ParameterTypes;
```

... where `ParameterTypeInfo` is a `readonly struct` containing the .NET type, the JS type string (see `JSComponentParameter.GetJSType`, and maybe flags like "is eventcallback" and "is eventcallback with parameter", or perhaps the .NET type that corresponds to `parameterType.GetGenericTypeDefinition`.

This way:

 * We don't need the JS-side code to make decisions based on whether the JS code supplies a function instance. Instead it can look at the type string supplied to it by .NET
 * We don't need the .NET-side code to make decisions by matching types during every time the parameters are set, but rather determines this info once up front
 * There's only one place to get/decide all this info